packages/frontend/src/api/queries.ts

```diff
@@ -19,9 +19,111 @@ import {
   type UpdateSystemLabelMutation,
   getSdk,
 } from 'bff-types-generated';
-import { GraphQLClient } from 'graphql-request';
+import { GraphQLClient, type RequestMiddleware, type ResponseMiddleware } from 'graphql-request';
+import { Analytics } from '../analytics';
+
+const requestMiddleware: RequestMiddleware = (request) => {
+  try {
+    // Extract operation name from the request
+    const operationName = request.operationName || 'UnknownOperation';
+    const startTime = Date.now();
+
+    // Store tracking data in request headers
+    if (!request.headers) {
+      request.headers = {};
+    }
+
+    // Add tracking headers
+    (request.headers as Record<string, string>)['x-graphql-operation'] = operationName;
```

Comment:
This seems a bit heavy, couldn't these variables (and such) be derived from a correlated request from BFF (proxy) to Dialogporten?